src/net.cpp

```diff
@@ -92,9 +92,9 @@ std::string strSubVersion;
 
 vector<CNode*> vNodes;
 CCriticalSection cs_vNodes;
-map<CInv, CDataStream> mapRelay;
-deque<pair<int64_t, CInv> > vRelayExpiration;
-CCriticalSection cs_mapRelay;
+map<uint256, CTransaction> mapRelayTx;
+deque<pair<int64_t, uint256> > vRelayTxExpiration;
+CCriticalSection cs_mapRelayTx;
```

Comment:
Perhaps we should we also change `mapAlreadyAskedFor` to be a `limitedmap<uint256, int64_t>`?  In the current implementation, we could have 2 requests in flight for the same txid, one for the witness version and one for the non-witness version, if we an upgraded- and non-upgraded-node both inv'ed the same transaction.